### PR TITLE
Dropbox(define methods)

### PR DESCRIPTION
**Pull request by advaypal**
Not fully functional, but thought it would be useful to have the methods clearly defined and merged in first. Will submit the fully functional PR asap

### CODE REVIEW

File: MIDIchlorians/MIDIchlorians/AppDelegate.swift [hunk 1]
```diff
@@ -47,12 +48,14 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     }
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+        //Initalise dropbox client
+        DropboxClientsManager.setupWithAppKey("8a5t7h2dpsfjunc")
```
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/AppDelegate.swift:52]: should we commit keys? :P

File: MIDIchlorians/MIDIchlorians/CloudManager.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    private func saveAudios(_ client: DropboxClient) {
+        guard let docsURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).last else {
+            return
+        }
+        //Get a list of samples
+        let samples = dataManager.loadAllAudioStrings()
+        //Upload each sample
+        for sample in samples {
+            let url = docsURL.appendingPathComponent("\(sample).\(Config.SoundExt)")
+            client.files.upload(path: "/\(Config.AudioFolderName)/\(sample).\(Config.SoundExt)", input: url)
```
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/CloudManager.swift:50]: eventually we might want to do something smarter? like checking if dropbox alr has the file :)

File: MIDIchlorians/MIDIchlorians/ViewController.swift [hunk 3]
```diff
@@ -142,6 +145,33 @@ class ViewController: UIViewController {
...
+                                                      UIApplication.shared.open(url) },
+                                                      browserAuth: false)
+    }
+
+    func importFromDropbox() {
+        if let client = DropboxClientsManager.authorizedClient {
+            cloudManager.loadFromDropbox(client: client)
+        } else {
+            loadDropBoxWebView()
+            importFromDropbox()
```
Comment (ngzhian) [inline MIDIchlorians/MIDIchlorians/ViewController.swift:162]: is this a recursive call? does it potentially loop for a long time?

**Review by ngzhian — COMMENTED** [review]
some clarifications needed

File: MIDIchlorians/MIDIchlorians/AppDelegate.swift (hunk 1 above)
Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/AppDelegate.swift:52]: Oh whoops

File: MIDIchlorians/MIDIchlorians/CloudManager.swift (hunk 2 above)
Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/CloudManager.swift:50]: Yup, will be in the next PR

File: MIDIchlorians/MIDIchlorians/ViewController.swift (hunk 3 above)
Comment (advaypal) [reply MIDIchlorians/MIDIchlorians/ViewController.swift:162]: My bad, this has the potential to loop indefinitely. Will change this

**Pull request closed without merging**